base/config_persistence.go

```diff
@@ -187,26 +184,19 @@ func (xbp *XattrBootstrapPersistence) loadConfig(ctx context.Context, c *gocb.Co
 			DebugfCtx(ctx, KeyCRUD, "No xattr config found for key=%s, path=%s: %v", key, cfgXattrConfigPath, xattrContErr)
 			return 0, ErrNotFound
 		}
-
-		// cas
-		var strCas string
-		xattrCasErr := res.ContentAt(1, &strCas)
-		if xattrCasErr != nil {
-			DebugfCtx(ctx, KeyCRUD, "No xattr cas found for key=%s, path=%s: %v", key, cfgXattrCasPath, xattrContErr)
-			return 0, ErrNotFound
-		}
-		cfgCas := HexCasToUint64(strCas)
+		casOut := res.Cas()
 
 		// deleted document check - if deleted, restore
 		var body map[string]interface{}
-		bodyErr := res.ContentAt(2, &body)
+		bodyErr := res.ContentAt(1, &body)
 		if bodyErr != nil {
-			restoreErr := xbp.restoreDocumentBody(c, key, valuePtr, strCas)
+			var restoreErr error
+			casOut, restoreErr = xbp.restoreDocumentBody(c, key, valuePtr, res.Cas())
```

Comment:
If there's an error here, I think `casOut` will be updated to 0, which might not be correct? The tombstone document will have a cas.

OTOH, the cas isn't super useful

Reply:
I can also see how if there's no document, then it should have a zero cas. I'm not sure what to do here.

Reply:
I think you're correct - if the restore fails we should return the cas of the current version of the document (the tombstone), since we're returning the tombstone's xattr.